squirrel.worker/src/test/java/org/dice_research/squirrel/fetcher/http/HTTPFetcherTest.java

```diff
@@ -0,0 +1,71 @@
+package org.dice_research.squirrel.fetcher.http;
+
+import org.dice_research.squirrel.*;
+import org.dice_research.squirrel.data.uri.CrawleableUri;
+import org.dice_research.squirrel.data.uri.CrawleableUriFactory;
+import org.dice_research.squirrel.data.uri.CrawleableUriFactoryImpl;
+import org.dice_research.squirrel.data.uri.UriType;
+import org.dice_research.squirrel.simulation.AbstractServerMockUsingTest;
+import org.dice_research.squirrel.simulation.CrawleableResource;
+import org.dice_research.squirrel.simulation.CrawleableResourceContainer;
+import org.dice_research.squirrel.simulation.StringResource;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.riot.Lang;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.simpleframework.http.core.Container;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+@RunWith(Parameterized.class)
+public class HTTPFetcherTest extends AbstractServerMockUsingTest {
+
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() throws Exception {
+        List<Object[]> scenarios = new ArrayList<Object[]>();
+        CrawleableUriFactory uriFactory = new CrawleableUriFactoryImpl();
+        String server1Url = "http://127.0.0.1:" + SERVER_PORT;
+
+        Model model1;
+
+        model1 = ModelFactory.createDefaultModel();
+        model1.add(model1.createResource(server1Url + "/entity_1.n3"),
+            model1.createProperty(server1Url + "/property_1.n3"), model1.createLiteral("literal"));
+        scenarios.add(new Object[]{
+            new CrawleableUri[]{uriFactory.create(new URI(server1Url + "/entity_1.n3"), UriType.DUMP)},
+            new CrawleableResource[]{new StringResource(model1, server1Url + "/entity_1.n3", Lang.N3),
+                new StringResource(model1, server1Url + "/property_1.n3", Lang.N3),
+                new StringResource(model1, server1Url + "/entity_2.n3", Lang.N3)}});
+        return scenarios;
+    }
+
+    private CrawleableUri[] seeds;
+    private CrawleableResource[] resources;
+
+    public HTTPFetcherTest(CrawleableUri[] seeds, CrawleableResource[] resources) {
+        super(new CrawleableResourceContainer(resources));
+        this.resources = resources;
+        this.seeds = seeds;
+    }
+
+
+
+    @Test
+    public void fetch() throws Exception {
+        int expected_file_content = 110;
+        HTTPFetcher fetcher = new HTTPFetcher();
+        File data = fetcher.fetch(seeds[0]);
```

Comment:
Why is the test generating crawleable URI objects for 4 classes but is checking only one single class?

Reply:
@MichaelRoeder  could you please elaborate on this comment.

Reply:
The strange name of the entities and property were misleading. I thought you are using four dump files but in the end you have 3 URIs returning the same content. However, this is still confusing. Why do you need all three resources if you are fetching only one of them?

Reply:
@MichaelRoeder I have removed the unnecessary fetching from two duplicate resources as it returns the same content and testing only one seed and one resource. Have committed the code to Testcase branch in abhihc/Squirrel (https://github.com/abhihc/Squirrel/commit/3640b16de4657ecda25b4c3be4552278a13a7ae4). Please let me know if I can push the code to abhihc/squirrel master so that reflects the code changes on this pull request.

Reply:
That already looks better. However, you added a second fetching step
```java
        //checks the crawlable uri object of second class
            File resourceData1 = fetcher.fetch(new CrawleableUri(new URI(resources[0].getResourceName())));
            assertNotNull(resourceData1);
            assertEquals(expected_file_content,resourceData1.length());
```
Can you please explain its meaning? What does the test checks here?

Reply:
@MichaelRoeder Similar to the first fetching step, I am checking from the resource as well, though it fetches the same data.

Reply:
Ok. Could you please explain the difference to the first fetching step? Maybe you want to (temporarily) add the following two lines to see what you are fetching:
```java
	System.out.println(seeds[0].getUri());
	System.out.println(resources[0].getResourceName());
```

Reply:
@MichaelRoeder Yes ok. Will add the above two lines to display what I am  fetching.

Reply:
@MichaelRoeder I get "http://127.0.0.1:8089/entity_1.n3"  for both the lines. I am fetching from this URI in both the cases.

Reply:
Exactly. So from my point if view, the second part does not really make sense since you are just repeating what you already did before. :wink: